lib/ansible/modules/cloud/openstack/os_security_group.py

```diff
@@ -87,28 +101,55 @@ def main():
         name=dict(required=True),
         description=dict(default=''),
         state=dict(default='present', choices=['absent', 'present']),
+        project=dict(default=None),
     )
 
     module_kwargs = openstack_module_kwargs()
     module = AnsibleModule(argument_spec,
                            supports_check_mode=True,
                            **module_kwargs)
 
+    if (module.params['project'] and
+            StrictVersion(shade.__version__) <= StrictVersion('1.21.0')):
```

Comment:
shade isn't gonna be here anymore - see recent changes to the module utils. we have an sdk module returned now. That said - 2.6 and beyond have a hard requirement on openstacksdk - so we can probably skip this version check.

Reply:
great, makes things much simpler.